packages/mui-utils/src/useIsFocusVisible.ts

```diff
@@ -102,7 +102,7 @@ export function teardown(doc: Document): void {
 function isFocusVisible(event: React.FocusEvent): boolean {
   const { target } = event;
   try {
-    return target.matches(':focus-visible');
+    return target.isEqualNode(document.activeElement);
```

Comment:
This is wrong, focus-visibel !== focus

```suggestion
    return target.matches(':focus-visible');
```

Reply:
true, though I believe this hook should be renamed or approached differently because current logic does not work in Safari

Reply:
> because current logic does not work in Safari

Do you have a reproduction? As far as I know, it works on Safari.

Reply:
yes, the bug is reproducible as described in the issue - the _:focus-visible_ implementation in Safari does not work as intended, besides, it hadn't been supported until recently

I believe my solution to be a bit more bug-proof for Safari and other browsers than this one